File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function(nums, k) {
    let s = new Set();

    for(let i=0;i<nums.length;i++) {
        if(s.has(nums[i]))
            return true;
        
        s.add(nums[i])
        if(s.size > k) {
            s.delete(nums[i-k])
        }
    }

    return false
};

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        hashmap = {}

        if k == 0:
            return False

        n = len(nums)
        l = 0
        for i in range(n):
            if i > k:
                hashmap[nums[l]] -= 1
                if hashmap[nums[l]] == 0:
                    del hashmap[nums[l]]
                l += 1

            if nums[i] in hashmap:
                return True
            else:
                hashmap[nums[i]] = 1
        
        return False